### DISCUSSION TITLE
1.3: Other formats links on HTML of book point to index.pdf and index.epub, how to change that?

### DISCUSSION THREAD

**Discussion by fuhrmanator** [quarto-dev/quarto-cli#3866] [Q&A] [answered]
Happy 2023!

My book project is now using Quarto 1.3 and I saw that by default the web page has "other formats" links to the PDF and Epub. That's practical:

![image](https://user-images.githubusercontent.com/7606540/210873481-f387ede3-2509-43a6-99a9-ad70959918a0.png)

However, they're broken links because they point to `index.pdf` and `index.epub`, which are not the files generated in my _quarto.yml.

The generated PDF is `Analyse-et-conception-de-logiciels.pdf` and I suspect the filename is automagically derived from the title:

```yml
book:
  title: "Analyse et conception de logiciels"
```

How can I fix this? I couldn't find anything in the documentation. 

ETA: my project is at https://github.com/fuhrmanator/log210-ndc-quarto and built to https://fuhrmanator.github.io/log210-ndc-quarto/

**Comment by dragonstyle** [comment, marked as answer]
I know that you're using a slightly older version of 1.3 due to a regression in later 1.3 builds, but unfortunately, this is a bug that is fixed in _later_ builds... If you update to the latest version of 1.3.86 (available now), it should contain fixes for the regression and also this issue. Sorry for the instability, it was unintentional but we're adding tests as we catch these cases that we've missed to try to prevent this in the future!

**Reply by fuhrmanator** [reply]
Thanks! I installed 1.3.86 on my PC and set it on the CI - there are some warnings about labels in my PlantUML captions, but I'll look into that (not a big worry). 

The downloads are working, although I set the option for them to be in the left side bar:

```yaml
book:
  repo-actions: [issue]
  downloads: [pdf, epub]
```

Thanks so much for your quick fix and understanding of my context. Cheers!

**Reply by fuhrmanator** [reply]
@dragonstyle I have more feedback (not sure if you want a bug report).

Even though I installed 1.3.86 on Windows, I only tried it on the CI (which is Ubuntu). On Windows, when I try to preview a PDF, it crashes with:

```
$ quarto preview c:/Users/Me/Documents/GitHub/log210-ndc-quarto/AnalyseConception.qmd --to pdf --no-browser --no-watch-inputs
...
running latexmk - 1
  Rc files read:
    NONE
  Latexmk: Run number 1 of rule 'pdflatex'
  This is pdfTeX, Version 3.141592653-2.6-1.40.24 (MiKTeX 22.11.25) (preloaded format=pdflatex.fmt)
   restricted \write18 enabled.
  entering extended mode
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error (11960): Illegal character ')'
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is damaged - attempting to reconstruct xref table...
  Syntax Error: Couldn't read xref table
  Syntax Warning: PDF file is da
compilation failed- missing packages (automatic installed disabled)
see C:\Users\Cris\Documents\GitHub\log210-ndc-quarto\index.log for more information.
```

This occurs even if I use `pdflatex` rather than `latexmk`.

There is a bug with (I believe) cross references inside of callouts, e.g.,

```markdown
::: {.callout-warning}
Attention de ne pas faire l'erreur naïve d'utiliser une classe de description simplement pour "décrire" une autre classe. Voir la @fig-MDD-ErreurDescription pour un exemple.
:::
```
renders as:
![image](https://user-images.githubusercontent.com/7606540/210928084-6b0ab088-8ea0-40fe-b6ef-d1863aec3d43.png)

There are warnings during the quarto build process, e.g., https://github.com/fuhrmanator/log210-ndc-quarto/actions/runs/3850656434/jobs/6561044069#step:11:48

I didn't try 1.3.88. Cheers!

**Reply by dragonstyle** [reply]
Ok this seems very possible due to some work we're doing internally around our LUA filters. I'll bisect and get to the bottom of this!

**Reply by dragonstyle** [reply]
Hmmm, I just rendered `log210-ndc-quarto` locally and didn't run into the issue with corruption. I did see these three warnings:

```
[WARNING] Citeproc: citation fig-LucidchartUML not found
[WARNING] Citeproc: citation fig-MDD-ErreurDescription not found
[WARNING] Citeproc: citation fig-dssdetails not found
```

which I will take a look at.

Are there LaTeX intermediates or anything else that might be causing the issue on your machine but not on a clean enlistment of the repo?

**Reply by dragonstyle** [reply]
There is indeed an issue with cross references inside callouts, opening an issue for that.

**Reply by fuhrmanator** [reply]
> Hmmm, I just rendered `log210-ndc-quarto` locally and didn't run into the issue with corruption. I did see these three warnings:
> 
> ```
> [WARNING] Citeproc: citation fig-LucidchartUML not found
> [WARNING] Citeproc: citation fig-MDD-ErreurDescription not found
> [WARNING] Citeproc: citation fig-dssdetails not found
> ```
> 
> which I will take a look at.
> 
> Are there LaTeX intermediates or anything else that might be causing the issue on your machine but not on a clean enlistment of the repo?

I updated to 1.3.90 this AM and the errors/warnings still show in Windows. However, quarto finally generates a (corrupt) PDF. The problems seem to be with the figures. I'm using MiKTeX on my Windows 10, from Visual Studio Code, in a bash shell (gitbash), not CMD.exe, if that matters. 

So far, the rater (bitmap) images are OK. It's the SVG and PlantUML (SVG) figures that don't render properly. I get errors in the viewer saying certain pages can't be rendered. Here are two figures that show trouble (the first is a filter-generated SVG from PlantUML, the second is a SVG done in Inkscape):

![image](https://user-images.githubusercontent.com/7606540/211066519-2ceb3059-ae73-4d9e-8472-5ae9af62e9b2.png)

![image](https://user-images.githubusercontent.com/7606540/211066555-3b9c06f7-8d6b-45bc-be61-b391aa50d8a4.png)

**Reply by dragonstyle** [reply]
Was this working in previous versions of Quarto? The fact that it is affecting SVG images is making me wonder if the issue has to do with the environment (for example not having `rsvg-convert` available).

**Reply by fuhrmanator** [reply]
> There is indeed an issue with cross references inside callouts, opening an issue for that.

FYI it appears that 1.3.91 (running on the CI of my book) fixes this problem. Thanks for the quick fix (if it's indeed that).

**Reply by cscheid** [reply]
Yes, that was a bug I introduced with a (large, under-the-hood) change in how quarto processes documents. Glad to hear it fixed the problem! (@dragonstyle: this was an AST bug.)

**Comment by fuhrmanator** [comment]
Yes it was working in 1.2, prior to the bug with duplicate svgs (discussed
in thread about corrupt PDFs). It also worked in the early version of 1.3,
but broke when I moved to the 1.3 version that fixed the download links
mentioned in this discussion.

**Reply by fuhrmanator** [reply]
To be specific (I was away from my PC before): it was 1.2.233 that worked on Windows with latexmk and svg files. 

It's 1.3.24 that works (in Windows) for me, and so I've rolled back to it, leaving the more recent version on the CI. Since the web site and PDF download are built by CI (which uses linux) I can live with this compromise for the time being (just need to check PDF changes locally as I make them in branches of the book project).
